cmd/tools/genvisitor/main.go

```diff
@@ -0,0 +1,68 @@
+package main
```

Comment:
can we have a section about the tool in `CONTRIBUTING.md` and how the repair works at high level?

Reply:
actually, probably best is to create an github issue for this work so we can record design choices.

Reply:
I added the contributing section (not a github issue)